CONTRIBUTING.md

```diff
@@ -30,3 +132,49 @@ Use culturally neutral language:
 
   What you may consider well-known may be entirely obscure and distracting to people from different backgrounds.
 
+### Licensing and attribution
+
+When opening pull requests with your own contributions, you agree to licensing your work under [CC-BY-SA 4.0][cc-by-sa].
+Before merging your work, you have to sign the [contributor agreement](cla/README.md).
+
+Having a single legal entity hold non-exclusive copyright avoids disputes and ensures the material can be put to use more effectively, e.g. by eventually publishing it as a book.
+You will still be considered co-author, as recorded by version history.
+
+When adding material by third parties, make sure it has a matching license that permits this.
+In that case, [unambiguously](#links) state source, authors, and license.
+Also [add the original author as co-author][github-multiple-authors] to the respective change, so we can track authorship through version history.
+
+Notify the authors *before* using their work.
+
+[cc-by-sa]: https://creativecommons.org/licenses/by-sa/4.0/
+[github-multiple-authors]: https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors
+
+### Links
+
+Unless explicitly required to point to the latest version of an external resource, all references should be [permanent links][permalink].
+
+Many web services offer permalinks.
+Examples:
+- [GitHub URLs to specific commits][github-permalink]
+- [Wikipedia URLs to specific page versions][wikipedia-permalink]
+- [Internet Archive "Save Page Now" for persisting web pages][internet-archive-save]
+
+[permalink]: https://en.m.wikipedia.org/wiki/Permalink
+[github-permalink]: https://docs.github.com/en/repositories/working-with-files/using-files/getting-permanent-links-to-files
+[wikipedia-permalink]: https://en.m.wikipedia.org/wiki/Wikipedia:Linking_to_Wikipedia#Permanent_links_to_old_versions_of_pages
+[internet-archive-save]: https://web.archive.org/save
```

Comment:
Lets take this as an example:

```suggestion
- [GitHub URLs to specific commits](https://docs.github.com/en/repositories/working-with-files/using-files/getting-permanent-links-to-files)
- [Wikipedia URLs to specific page versions](https://en.m.wikipedia.org/wiki/Wikipedia:Linking_to_Wikipedia#Permanent_links_to_old_versions_of_pages)
- [Internet Archive "Save Page Now" for persisting web pages](https://web.archive.org/save)
```
This is what I would write normally and I would prefer here. It is shorter and easier to know what links where.